src/main/java/cstwooneohthree/glados/tasks/Task.java

```diff
@@ -0,0 +1,94 @@
+package cstwooneohthree.glados.tasks;
+
+import cstwooneohthree.glados.enums.TaskType;
+
+/**
+ * Abstract task class that handles whether it is done or not.
+ *
+ * @author jayjay19630
+ */
+public abstract class Task {
+
+    /* Description and boolean for task being done */
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs task and sets description.
+     * By default not marked as done.
+     *
+     * @param description
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Return task type icon.
+     *
+     * @return Icon in the form [...].
+     */
+    public abstract String getTaskTypeIcon();
+
+    /**
+     * Return task type enum.
+     *
+     * @return Task enum.
+     */
+    public abstract TaskType getTaskType();
+
+    /**
+     * Return task description.
+     *
+     * @return Task description.
+     */
+    public abstract String getDescription();
+
+    /**
+     * Return task description to be saved in data file.
+     *
+     * @return Task description in save format.
+     */
+    public abstract String getSaveFormat();
```

Comment:
I really like that your method names in general are phrased as verbs and detail their functions clearly!